R/write_metadata.R

```diff
@@ -30,9 +30,45 @@
     write(x = json, file = path)
 }
 
+#' Read the .zattrs file associated with a Zarr array or group
+#' 
+#' @param path A character vector of length 1. This provides the
+#'   path to a Zarr array or group. 
+#' @param new.zattrs a list inserted to .zattrs at the \code{path}.
+#' @param overwrite if TRUE, existing .zattrs elements will be overwritten by \code{new.zattrs}.
+#' 
+#' @importFrom jsonlite toJSON
+#'
+#' @export
+write_zattrs <- function(path, new.zattrs = list(), overwrite = TRUE){
+  path <- .normalize_array_path(path)
+  zattrs_path <- paste0(path, ".zattrs")
+  
+  if(is.null(names(new.zattrs)))
+    stop("list elements should be named")
+  
+  if("" %in% names(new.zattrs)){
+    message("Ignoring unnamed list elements")
+    new.zattrs <- new.zattrs[which(names(new.zattrs == ""))]
```

Comment:
I don't think this is what you intended. You are checking if the value is "", rather than the name. Also, it's probably supposed to be `!=` rather than `==`.

```suggestion
    new.zattrs <- new.zattrs[which(names(new.zattrs) != "")]
```

Or better


```suggestion
    new.zattrs <- new.zattrs[nzchar(names(new.zattrs))]
```

Could you also add a test that would have caught this please?